Framework/Kernel/inc/MantidKernel/PropertyWithValue.h

```diff
@@ -637,6 +641,29 @@ template <typename TYPE> class DLLExport PropertyWithValue : public Property {
   PropertyWithValue();
 };
 
+#define PROPERTYWITHVALUE_SAVEPROPERTY(type)                                   \
+  template <>                                                                  \
+  inline void PropertyWithValue<type>::saveProperty(::NeXus::File *file) {     \
+    file->makeGroup(this->name(), "NXlog", 1);                                 \
+    file->writeData("value", m_value);                                         \
+    file->closeGroup();                                                        \
+  }
+
+PROPERTYWITHVALUE_SAVEPROPERTY(float)
+PROPERTYWITHVALUE_SAVEPROPERTY(double)
+PROPERTYWITHVALUE_SAVEPROPERTY(int32_t)
+PROPERTYWITHVALUE_SAVEPROPERTY(uint32_t)
+PROPERTYWITHVALUE_SAVEPROPERTY(int64_t)
+PROPERTYWITHVALUE_SAVEPROPERTY(uint64_t)
+PROPERTYWITHVALUE_SAVEPROPERTY(std::string)
+PROPERTYWITHVALUE_SAVEPROPERTY(std::vector<double>)
+PROPERTYWITHVALUE_SAVEPROPERTY(std::vector<int32_t>)
+
+template <typename TYPE>
+inline void PropertyWithValue<TYPE>::saveProperty(::NeXus::File *file) {
+  this->saveProperty(file);
+}
```

Comment:
I made the smaller changes you suggested and moved some of the template specializations out of the header and into the implementation. Before merging into master, I do want to cleanup `PropertyWithValue<>::saveProperty(...)`  (possibly using std::enable_if and increasing the number of instantiations).

We may want to keep an implementation of saveProperty in the header because there are many specializations of PropertyWithValue that don't work with saveProperty. Listing all of them seems to defeat some benefits of this being a templated class.